contracts/ExampleNFT-v2.cdc

```diff
@@ -0,0 +1,425 @@
+                    traitsView.addTrait(mintedTimeTrait)
+
+                    // foo is a trait with its own rarity
+                    let fooTraitRarity = MetadataViews.Rarity(score: 10.0, max: 100.0, description: "Common")
+                    let fooTrait = MetadataViews.Trait(name: "foo", value: self.metadata["foo"], displayType: nil, rarity: fooTraitRarity)
+                    traitsView.addTrait(fooTrait)
+                    
+                    return traitsView
+
+            }
+            return nil
+        }
+    }
+
+    pub resource Collection: NonFungibleToken.NFTCollection, NonFungibleToken.Provider, NonFungibleToken.Receiver, NonFungibleToken.Transferor, NonFungibleToken.CollectionPublic, MetadataViews.ResolverCollection {
+        /// dictionary of NFT conforming tokens
+        /// NFT is a resource type with an `UInt64` ID field
+        access(contract) var ownedNFTs: @{UInt64: ExampleNFT.NFT{NonFungibleToken.NFT}}
+
+        access(self) var storagePath: StoragePath
+        access(self) var publicPath: PublicPath
+
+        /// Return the default storage path for the collection
+        pub fun getDefaultStoragePath(): StoragePath? {
+            return self.storagePath
+        }
+
+        /// Return the default public path for the collection
+        pub fun getDefaultPublicPath(): PublicPath? {
+            return self.publicPath
+        }
+
+        init () {
+            self.ownedNFTs <- {}
+            let identifier = "cadenceExampleNFTCollection"
+            self.storagePath = StoragePath(identifier: identifier)
+            self.publicPath = PublicPath(identifier: identifier)
+        }
+
+        /// Returns the NFT types that this collection can store
+        pub fun getAcceptedTypes(): {Type: Bool} {
+            return {
+                Type<@ExampleNFT.NFT>(): true
+            }
+        }
+
+        /// withdraw removes an NFT from the collection and moves it to the caller
+        pub fun withdraw(withdrawID: UInt64): @ExampleNFT.NFT{NonFungibleToken.NFT} {
+            let token <- self.ownedNFTs.remove(key: withdrawID)
+                ?? panic("Could not withdraw an NFT with the provided ID from the collection")
+
+            emit Withdraw(id: token.id, from: self.owner?.address, type: token.getType(),
+                          displayView: token.resolveView(MetadataViews.Type<MetadataViews.Display>()),
+                          serialView: token.resolveView(MetadataViews.Type<MetadataViews.Serial>()))
+
+            return <-token
+        }
+
+        /// deposit takes a NFT and adds it to the collections dictionary
+        /// and adds the ID to the id array
+        pub fun deposit(token: @AnyResource{NonFungibleToken.NFT}) {
+            let token <- token as! @ExampleNFT.NFT
+
+            emit Deposit(id: token.id, to: self.owner?.address, type: token.getType(),
```

Comment:
Events emitted in interface?

Reply:
We're gonna discuss that in the breakout session today. We need to make a FLIP for that, but I need help because I'm not sure what we would put in the FLIP